src/proposals/mips/mip-rewards-distribution.sol

```diff
@@ -0,0 +1,510 @@
+                ActionType.Moonbeam
+            );
+
+            uint16 wormholeChainId = bridgeWell.network.toWormholeChainId();
+
+            uint256 bridgeCost = xWELLRouter(router).bridgeCost(
+                wormholeChainId
+            );
+
+            _pushAction(
+                router,
+                bridgeCost,
+                abi.encodeWithSignature(
+                    "bridgeToRecipient(address,uint256,uint16)",
+                    target,
+                    bridgeWell.amount,
+                    wormholeChainId
+                ),
+                string(
+                    abi.encode(
+                        "Bridge %s WELL to %s on chain %s",
+                        bridgeWell.amount,
+                        target,
+                        bridgeWell.network
+                    )
+                ),
+                ActionType.Moonbeam
+            );
+        }
+
+        _pushAction(
+            addresses.getAddress("STK_GOVTOKEN"),
+            abi.encodeWithSignature(
+                "configureAsset(uint128,address)",
+                spec.stkWellEmissionsPerSecond,
+                addresses.getAddress("STK_GOVTOKEN")
+            ),
+            "Set reward speed for the Safety Module on Moonbeam",
+            ActionType.Moonbeam
+        );
+
+        for (uint256 i = 0; i < spec.setRewardSpeed.length; i++) {
+            SetRewardSpeed memory setRewardSpeed = spec.setRewardSpeed[i];
+
+            _pushAction(
+                addresses.getAddress(setRewardSpeed.target),
+                abi.encodeWithSignature(
+                    "_setRewardSpeed(uint8, address, uint256,uint256)",
+                    uint8(setRewardSpeed.rewardType),
+                    addresses.getAddress(setRewardSpeed.market),
+                    setRewardSpeed.newBorrowSpeed,
+                    setRewardSpeed.newSupplySpeed
+                ),
+                "Set reward speed for the Moonwell Markets on Moonbeam",
+                ActionType.Moonbeam
+            );
+        }
+
+        AddRewardInfo memory addRewardInfo = spec.addRewardInfo;
+
+        // first approve amount
+        _pushAction(
+            addresses.getAddress("GOVTOKEN"),
+            abi.encodeWithSignature(
+                "approve(address,uint256)",
+                addresses.getAddress(addRewardInfo.target),
+                addRewardInfo.amount
+            ),
+            "Approve StellaSwap spend the amount of WELL",
+            ActionType.Moonbeam
+        );
+
+        _pushAction(
+            addresses.getAddress(addRewardInfo.target),
+            abi.encodeWithSignature(
+                "addRewardInfo(uint256,uint256,uint256)",
+                addRewardInfo.pid,
+                addRewardInfo.endTimestamp,
+                addRewardInfo.rewardPerSec
+            ),
+            "Add reward info for the Moonwell Markets on Moonbeam",
+            ActionType.Moonbeam
+        );
```

Comment:
@anajuliabit if you want I can pass in the start timestamp and end timestamp of the new epoch, but it might be easier for me to just pass in the total amount.